ams/ams-server/src/main/java/com/netease/arctic/ams/server/service/impl/OptimizeQueueService.java

```diff
@@ -484,11 +484,16 @@ public String queueName() {
 
     public OptimizeTask poll(JobId jobId, final String attemptId, long waitTime) {
       long startTime = System.currentTimeMillis();
+      boolean first = true;
       while (true) {
-        long duration = System.currentTimeMillis() - startTime;
-        if (duration > waitTime) {
-          LOG.warn("pool task cost too much time {} ms, return null", duration);
-          return null;
+        if (first) {
+          first = false;
+        } else {
+          long duration = System.currentTimeMillis() - startTime;
+          if (duration > waitTime) {
```

Comment:
I suggest putting the timeout judgment at the end of the method. In addition, the current method body is too long. It is recommended to abstract some complex logic into private methods to shortage this method.

Reply:
OK, I will refactor it.